driver/src/test/java/org/neo4j/driver/TransactionConfigTest.java

```diff
@@ -118,4 +118,38 @@ void shouldHaveMetadata()
         assertEquals( value( true ), metadata.get( "key2" ) );
         assertEquals( value( 42 ), metadata.get( "key3" ) );
     }
+
+    @Test
+    void cannotMessWithAMapAfterPassingItToTheBuilder() {
```

Comment:
To keep naming consistent with the majority of other tests, should we name it something like: `shouldNotModifyMetadataAfterItIsSuppliedToBuilder`?